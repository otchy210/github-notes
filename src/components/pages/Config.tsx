import React from 'react';
import { useAccessToken } from '../../hooks/useAccessToken';

export const Config: React.FC = () => {
  const [accessToken, setAccessToken] = useAccessToken();
  return (
    <>
      <h2>GitHub Personal access token</h2>
      <p>
        You can create new access token on{' '}
        <a href="https://github.com/settings/tokens" target="_blank">
          the GitHub
        </a>
        . You need to give "repo" scope to use this tool. The secret will be stored on your browser so no one except you can check the value.
      </p>
      <p>
        <input
          type="password"
          value={accessToken}
          onChange={(e) => {
            setAccessToken(e.target.value);
          }}
        />
      </p>
    </>
  );
};
