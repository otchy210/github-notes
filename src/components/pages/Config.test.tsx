import React from 'react';
import { render } from '@testing-library/react';
import { Config } from './Config';

describe('Config', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Config />);
    expect(asFragment()).toMatchSnapshot();
  });
});
