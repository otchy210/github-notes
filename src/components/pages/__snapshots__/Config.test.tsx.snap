// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    GitHub Personal access token
  </h2>
  <p>
    You can create new access token on 
    <a
      href="https://github.com/settings/tokens"
      target="_blank"
    >
      the GitHub
    </a>
    . You need to give "repo" scope to use this tool. The secret will be stored on your browser so no one except you can check the value.
  </p>
  <p>
    <input
      type="password"
      value=""
    />
  </p>
</DocumentFragment>
`;
